templates/commands/render_flags.go

```diff
@@ -51,6 +51,9 @@ type RenderFlags struct {
 	// Inputs provide values that are substituted into the template. The keys in
 	// this map must match the input names in the Source template's spec.yaml
 	// file.
+	//
+	// This is mutable, even after flag parsing. It may be updated when default
+	// values are added and as the user is prompted for inputs.
```

Comment:
For the future: I wonder if we need a way to indicate that inputs are "finalized". I can't think of a good reason right now.

Reply:
I think the way I'd approach this more cleanly is to have this variable be immutable after parsing, and then keep track of the inputs-with-defaults-and-prompting as just a local variables. I just want to avoid having long-lived global-esque mutable state. But probably not needed right now, like you say.